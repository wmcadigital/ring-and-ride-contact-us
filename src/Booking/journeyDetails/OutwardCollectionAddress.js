import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutJourneySectionName from "./getAboutJourneySectionName";

import { required } from "../../common/validation";

const OutwardCollectionAddress = ({
  setOutwardCollectionAddress,
  setGoToPage,
}) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const bookingParty = formValues["bookingParty"];

  const error = stateApi.submitFailed
    ? stateApi.errors?.outwardCollectionAddress
    : null;

  const question =
    bookingParty === "mySelf"
      ? "Where are you being collected from?"
      : `Where is ${formValues["firstName"]} being collected from?`;

  const labelOne =
    bookingParty === "mySelf"
      ? "My registered home address"
      : "Their registered home address";

  useEffect(() => {
    if (formValues.outwardCollectionAddress) {
      setOutwardCollectionAddress(formValues.outwardCollectionAddress);
    }
    if (formValues.outwardCollectionAddress === "other") {
      setGoToPage(null);
    }
  }, [formValues.outwardCollectionAddress]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("2", stateApi)}
        sectionName={getAboutJourneySectionName(stateApi)}
      />
      <Question text={question} />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label={labelOne}
          validation={required}
          value="registered"
          fieldName="outwardCollectionAddress"
        />
        <RadioButton
          key={2}
          label="Another address"
          validation={required}
          value="other"
          fieldName="outwardCollectionAddress"
        />
      </RadioGroup>
    </FormSection>
  );
};

OutwardCollectionAddress.propTypes = {
  setOutwardCollectionAddress: PropTypes.func,
  setGoToPage: PropTypes.func,
};

OutwardCollectionAddress.defaultProps = {
  setOutwardCollectionAddress: () => {},
  setGoToPage: () => {},
};

export default OutwardCollectionAddress;
