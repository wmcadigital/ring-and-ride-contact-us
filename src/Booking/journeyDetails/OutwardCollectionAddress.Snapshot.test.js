import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import OutwardCollectionAddress from "./OutwardCollectionAddress";

jest.mock("react-final-form");

describe("OutwardCollectionAddress - snapshots", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<OutwardCollectionAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<OutwardCollectionAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { outwardCollectionAddress: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<OutwardCollectionAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
