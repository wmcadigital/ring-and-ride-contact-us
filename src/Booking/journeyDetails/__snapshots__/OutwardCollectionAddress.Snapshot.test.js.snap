// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutwardCollectionAddress - snapshots individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Where is John being collected from?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="Their registered home address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="registered"
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="Another address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="other"
    />
  </RadioGroup>
</FormSection>
`;

exports[`OutwardCollectionAddress - snapshots individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Where are you being collected from?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="My registered home address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="registered"
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="Another address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="other"
    />
  </RadioGroup>
</FormSection>
`;

exports[`OutwardCollectionAddress - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="Where are you being collected from?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="My registered home address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="registered"
    />
    <RadioButton
      fieldName="outwardCollectionAddress"
      label="Another address"
      onClickCallback={[Function]}
      validation={[Function]}
      value="other"
    />
  </RadioGroup>
</FormSection>
`;
